FilePersistence/src/version_control/Commit.cpp

```diff
@@ -129,24 +129,26 @@ bool Commit::isValidMatch(QString content, int indexOfId, int& start, int& end,
 
 QStringList Commit::nodeLinesFromId(Model::NodeIdType id, bool findChildrenByParentId) const
 {
-	auto idText =  id.toString();
+	auto idText =  id.toString().toLatin1();
 	QStringList matches;
 	for (auto file : files())
 	{
-		int indexOfId = 0;
-		auto content = QString::fromUtf8(file->content());
-
-		indexOfId = content.indexOf(idText, indexOfId);
-		if (indexOfId != -1)
+		const char *matchingLocation = strstr(file->content(), idText.constData());
+		while (matchingLocation)
 		{
 			int start, end;
-			if (isValidMatch(content, indexOfId, start, end, findChildrenByParentId))
+			if (isValidMatch(file->content(), file->size_, matchingLocation, start, end, findChildrenByParentId))
 			{
-				QString match = file->relativePath_ + ":" + content.mid(start, end-start);
+				char *data = new char[end-start+1];
+				strncpy(data, file->content()+start, end-start);
+				data[end-start]='\0';
+				QString match = file->relativePath_ + ":" + data;
+				delete data;
```

Comment:
Oh, yes of course. Thanks @lukedirtwalker.

@Vaishal-shah: could you please submit a PR fixing this? However instead of simply putting the missing `[]`, let's make the code better by using `QByteArray` that will take care of memory management:

    QString match{file->relativePath_};
    match << QByteArray{file->content()+start, end-start};